src/main.rs

```diff
@@ -75,3 +76,148 @@ async fn main() {
     )
     .await;
 }
+
+#[cfg(test)]
+mod tests {
+
+    use super::*;
+    use dockertest::waitfor::{MessageSource, MessageWait};
+    use dockertest::{DockerTest, Image, TestBodySpecification};
+    use std::env;
+    use std::process::Command;
+    use std::sync::{Arc, Mutex};
+    use tokio::time::{sleep, Duration};
+
+    const MAGIC_LINE: &str = "postgres password: ";
+
+    fn quasar_database_spec() -> TestBodySpecification {
+        let exit_wait = MessageWait {
+            message: "database system is ready to accept connections".to_string(),
+            source: MessageSource::Stderr,
+            timeout: 5,
+        };
+
+        let mut quasar_proc = TestBodySpecification::with_repository("postgres")
+            .set_wait_for(Box::new(exit_wait))
+            .set_handle("quasar");
+
+        quasar_proc.modify_port_map(5432, 5432);
```

Comment:
It might be worth considering passing some of these values as env rather than in code. Eg, as a dev, I might want to run tests but use a different port for postgres.